apache-rat-core/src/main/java/org/apache/rat/document/DocumentNameMatcher.java

```diff
@@ -20,17 +20,206 @@
 
 import java.io.File;
 import java.io.FileFilter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
 
-@FunctionalInterface
-public interface DocumentNameMatcher {
-    boolean matches(DocumentName documentName);
+import org.apache.rat.config.exclusion.plexus.MatchPattern;
+import org.apache.rat.config.exclusion.plexus.MatchPatterns;
+
+import static java.lang.String.format;
+
+/**
+ * Matches document names.
+ */
+public final class DocumentNameMatcher {
+
+    /** The environment variable to set to enable tracing for the PatternMatcher execution */
+    public static final String ENV_VAR = DocumentNameMatcher.class.getName();
```

Comment:
Shouldn't there be a reference to this ENV_VAR if it is used to indicate or change something?